graphql-dgs-subscriptions-sse/src/main/kotlin/com/netflix/graphql/dgs/subscriptions/sse/DgsSSESubscriptionHandler.kt

```diff
@@ -44,119 +49,73 @@ import java.util.*
-                    ).id(UUID.randomUUID().toString())
-                emitter.send(event)
-
-                subscription.request(1)
-            }
-
-            override fun onError(t: Throwable) {
-                logger.error("Error on subscription {}", sessionId, t)
-                val event = SseEmitter.event()
-                    .data(
-                        mapper.writeValueAsString(
-                            SSEDataPayload(
-                                data = null,
-                                errors = listOf(Error(t.message)),
-                                subId = sessionId
-                            )
-                        ),
-                        MediaType.APPLICATION_JSON
-                    )
-
-                emitter.send(event)
-                emitter.completeWithError(t)
-            }
-
-            override fun onComplete() {
-                emitter.complete()
-            }
-        }
-
-        emitter.onError {
-            logger.warn("Subscription {} had a connection error", sessionId)
-            subscriber.subscription.cancel()
-        }
-
-        emitter.onTimeout {
-            logger.warn("Subscription {} timed out", sessionId)
-            subscriber.subscription.cancel()
+            logger.error(errorMessage)
+            throw ServerWebInputException(errorMessage)
         }
 
         val publisher = try {
             executionResult.getData<Publisher<ExecutionResult>>()
-        } catch (ex: ClassCastException) {
-            return if (query.contains("subscription")) {
-                logger.error("Invalid return type for subscription datafetcher. A subscription datafetcher must return a Publisher<ExecutionResult>. The query was $query", ex)
-                emitter.send("Invalid return type for subscription datafetcher. Was a non-subscription query send to the subscription endpoint?")
-                emitter.complete()
-                ResponseEntity.status(500).body(emitter)
-            } else {
-                logger.warn("Invalid return type for subscription datafetcher. The query sent doesn't appear to be a subscription query: $query", ex)
-                emitter.send("Invalid return type for subscription datafetcher. Was a non-subscription query send to the subscription endpoint?")
-                emitter.complete()
-                ResponseEntity.badRequest().body(emitter)
-            }
+        } catch (exc: ClassCastException) {
+            logger.error(
+                "Invalid return type for subscription datafetcher. A subscription datafetcher must return a Publisher<ExecutionResult>. The query was {}",
+                query, exc
+            )
+            throw ServerErrorException("Invalid return type for subscription datafetcher. Was a non-subscription query send to the subscription endpoint?", exc)
         }
 
-        publisher.subscribe(subscriber)
+        val subscriptionId = UUID.randomUUID().toString()
```

Comment:
How does the execution Id get set here?

Reply:
I don't think we can get the ExecutionId here.